File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        maxLength = 0
        right, left = 0, 0
        freq = {}
        maxFreq = 0
        while right < len(s):
            freq[s[right]] = freq.get(s[right], 0) + 1
            maxFreq = max(maxFreq, freq[s[right]])
            if right - left + 1 - maxFreq > k:
                freq[s[left]] -= 1
                left += 1
            right += 1
        
        return right - left

File: Data Structures & Algorithms/is-palindrome/submission-4.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int left = 0;
        int right = s.size() - 1;
        while(left <= right) {
            if (!isalnum(s[left])) {
                left++;
            } else if (!isalnum(s[right])) {
                right--;
            } else {
                char l = tolower(s[left]);
                char r = tolower(s[right]);
                if (l != r) {
                    return false;
                }
                right--;
                left++;
            }
        }
        return true;
    }
};
